### ISSUE TITLE
Option to disable search auto completions in vomnibar

### ISSUE THREAD

**Issue by paramjitrohits** [philc/vimium#3078]
Hi,
I use vomnibar and use certain custom search engines as mentioned in the documentation. However, I do not like being spoonfed with autocomplete suggestions every time I want to make a specific query. Can an option be added to disable autocomplete suggestions? (I couldn't find any in the documentation) Thanks for the project!

**Comment by smblott-github** [comment]
Vimium will *always* do completion for a custom search engine if the URL for your custom search engine matches the [regex](https://github.com/philc/vimium/blob/master/background_scripts/completion_engines.coffee#L47) associated with Vimium's built-in completion engine.

Perhaps you could try using a URL which works, but which does not match the regex?

**Comment by paramjitrohits** [comment]
Hi,
I think the problem is that the regexes for custom search engines (take Google for instance) are very broad, and I can't really think of a way to query Google without using a url already covered by the completion engine regex. 

I see that Vimium always does completion as it stands, I was asking if there is a way to override this mechanism, or if such a functionality could be added.

**Comment by smblott-github** [comment]
> I was asking if there is a way to override this mechanism

Not currently.

> or if such a functionality could be added.

We'll just leave this issue here, and see if others support the idea.  It's pretty niche, though.

**Comment by paramjitrohits** [comment]
Thanks for the consideration. If one prefers to avoid getting distracted by trending stupid search queries or leaking queries about which one hasn't made one's mind, it starts to seem less niche.

**Comment by Geremia** [comment]
@smblott-github Which engines are built-in to Vimium?

**Comment by smblott-github** [comment]
Go to the options page and click on the "here" link bottom right in the screenshot...

![image](https://user-images.githubusercontent.com/2641335/50438467-f99f2400-08e5-11e9-96c1-ab4bcc8d1ad9.png)

**Comment by idontlikecreatingaccounts** [comment]
I would also like to disable the search engine suggestions.

Additionally, I wish there was an option to disable the history suggestions when typing in a new URL.

Both of these are basic features of any browser, and ones I always opt to use. It's a shame the vomnibar forces everyone to have suggestions pop up regardless of their browser settings.

**Comment by philc** [comment]
I've come around to thinking that it's reasonable to have an option to disable completions, for privacy reasons, or to denoise the UX. Unclear to me if this should be a global configuration, or per-search engine provider.

**Comment by gaesa** [comment]
Perhaps it could be both: a global default, with an option to override per search engine.
If no override is provided, it would fall back to the global setting.

**Comment by philg-dev** [comment]
I just had the idea, that it could work dynamically in a similar way to `bash` history setting `HISTCONTROL=ignorespace`: by prepending a ` ` (space) at the start of the search entry. In `bash` the command history doesn't save commands that start with a space character.

That way it would be in the user's control and they could decide on-the-fly which one they want depending on their situation (e.g. in public / screen-sharing vs. at home). There could be a global flag to flip this functionality:
- disable auto-completion by prepending space (default; least disruptive for existing users, as auto-completion would happen just as usual, except when prepending a space)
- enable auto-completion by prepending space

I don't think it's currently possible to define ` ` (space) as a search engine keyword, so I think space would be suitable independently of user configuration? Aside from that the prefix character for toggling auto-completion could also get a setting.

**Comment by idontlikecreatingaccounts** [comment]
> That way it would be in the user's control and they could decide on-the-fly which one they want depending on their situation (e.g. in public / screen-sharing vs. at home). There could be a global flag to flip this functionality:
> 
>     * disable auto-completion by prepending space (default; least disruptive for existing users, as auto-completion would happen just as usual, except when prepending a space)
> 
>     * enable auto-completion by prepending space

I understand the value of customizability and letting people have more control over their setup.

However, as for myself, and I'm sure some other people too, I simply never ever want these suggestions. Not on the address bar, not on the omnibar. I certainly don't want what's perhaps the most pressed key on the keyboard to be a "suggestion jumpscare" key.

So, while in theory I see the upside of an "on-the-fly" switch for those who want such functionality, I'd personally vastly prefer to just have a clean setting in the options to never have suggestions, regardless of what I type or paste in.

**Comment by shihiro09** [comment]
@idontlikecreatingaccounts, agreed. In addition, I think a global off-switch is the simplest to implement. I also think support for a global option is necessary just as a defensive approach: if I forget to add some kind of opt-out trigger (especially on the fly -- likely to happen), I don't want that engine keylogging me by default.

If users want more granularity later, maybe a design could be decided then? But right now, it's not clear whether there is demand for such granularity. Firefox's search engine completion is globally gated through `browser.search.suggest.enabled`, so there *is* browser-level precedent for a global toggle (though, technically speaking, firefox's custom engine creation does offer one more granular control).

So there's browser precedent for both, but the easiest to implement and reason about is most definitely a global option, which is probably necessary regardless which direction vimium decides to go with the per-engine options.
